backend/app/api/src/helpers/UitpasTokenRepository.ts

```diff
@@ -0,0 +1,102 @@
+import { SimpleError } from '@simonbackx/simple-errors';
+import { UitpasToken } from '@stamhoofd/models';
+
+export class UitpasTokenRepository {
+    accessToken?: string;
+    expiresOn: Date = new Date(0); // Set to minimum time initially
+    uitpasToken: UitpasToken;
+
+    constructor(uitpasToken: UitpasToken) {
+        this.uitpasToken = uitpasToken;
+    }
+
+    static knownTokens: Map<string | null, UitpasTokenRepository> = new Map(); // organizationId (null means platform) -> UitpasToken
+
+    static async getAccessTokenFor(organizationId: string | null = null, forceRefresh: boolean = false): Promise<string> {
+        const repo = UitpasTokenRepository.knownTokens.get(organizationId);
+        if (repo) {
+            return repo.getAccessToken(forceRefresh);
+        }
+        // query db
+        let uitpasToken = await UitpasToken.select().where('organizationId', organizationId).first(false);
+        if (!uitpasToken) {
+            // temporary solution, because platform client id and secret are not yet in the database
+            if (organizationId === null) {
+                if (!STAMHOOFD.UITPAS_API_CLIENT_ID || !STAMHOOFD.UITPAS_API_CLIENT_SECRET) {
+                    throw new SimpleError({
+                        code: 'uitpas_api_not_configured',
+                        message: 'UiTPAS API is not configured',
+                        human: $t('UiTPAS is niet volledig geconfigureerd. Neem contact op met de platformbeheerder.'),
+                    });
+                }
+                uitpasToken = new UitpasToken();
+                uitpasToken.clientId = STAMHOOFD.UITPAS_API_CLIENT_ID;
+                uitpasToken.clientSecret = STAMHOOFD.UITPAS_API_CLIENT_SECRET;
+                uitpasToken.organizationId = null; // null means platform
+            }
+            else {
+                throw new SimpleError({
+                    code: 'uitpas_token_not_found',
+                    message: organizationId ? `Uitpas token not found for clientId ${organizationId}` : 'Uitpas token not found for platform',
+                    human: $t(`De UiTPAS integratie is niet compleet, contacteer de beheerder.`),
+                });
+            }
+        }
+        const newRepo = new UitpasTokenRepository(uitpasToken);
+        this.knownTokens.set(organizationId, newRepo);
+        return newRepo.getAccessToken(true);
+    }
+
+    private async getAccessToken(forceRefresh: boolean = false): Promise<string> {
+        if (forceRefresh || !this.accessToken || this.expiresOn < new Date()) {
```

Comment:
Ik zou nog een 5tal seconden extra toevoegen bij de expiry check, zodat we de token al opnieuw opvragen als hij heel binnenkort vervalt - dat gaat vermijden dat hij net net vervalt en we op heel willekeurige momenten een error gooien